velox/dwio/common/ScanSpec.h

```diff
@@ -295,6 +295,13 @@ class ScanSpec {
     reorder();
   }
 
+  void clearHasFilter() {
```

Comment:
Can we do it this a little bit differently since `resetCachedValues` is only used in `addDynamicFilter`:
1. Remove `reorder` from `resetCachedValues`
2. Rename `reorder` to `reorderChildren` and make it public
3. Call `reorderChildren` after `resetCachedValues` in `addDynamicFilter`
4. Call `resetCachedValues` only in `addSplit`

Then we can add a unit test to make sure no body will break this.

Reply:
Thanks, I now know why the previous tests(DWRF) didn't reproduce the issue. It is a funny case.

In general, We will `setConstantValue` for partition key and call `scanChildSpec->setConstantValue(nullptr);` for another column(like id). 
When I use DWRF format. [fileType](https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L516) is `"ROW<c0:BIGINT>"` I need to make col name as c0(default) not id. otherwise,  logic will goto [setNullConstantValue](https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L535-L537.) not [scanChildSpec->setConstantValue(nullptr);](https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L539)

Then it will add a dwrf reader in SelectiveStructColumnReader [construct](https://github.com/facebookincubator/velox/blob/main/velox/dwio/dwrf/reader/SelectiveStructColumnReader.cpp#L50-L52) to call outputRows().

Reply:
Because ScanSpec has a function call `reorder`. So I change  the signature to `resetCachedValues(bool isReorder)` for compile.